### ISSUE TITLE
Phase 3: MLOps loop を実装する（CI / reproducibility / drift / uncertain queue）

### ISSUE THREAD

**Issue by yo4e** [yo4e/gaiden-sieve#6]
## 目的

Phase 2 で作った model lifecycle を土台に、GAIDEN SIEVE の v0 MLOps loop を一周させる。

この Phase では、production model を自動で書き換えることではなく、**コード・学習・評価が継続して再現できること、入力分布の変化を観測できること、判断保留の記事を次の学習候補として回収できること**を確認する。

```text
code + fixture/data
       ↓
 GitHub Actions
       ↓
test → train → evaluate → report
                         ↓
production model → batch classify
                         ↓
                    drift report
                         ↓
                  uncertain queue
```

### 1. GitHub Actions

- [ ] `.github/workflows/mlops.yml` を追加する
- [ ] pull request / push ごとに小さい fixture / test dataset で以下を実行する
  - [ ] install
  - [ ] pytest
  - [ ] candidate train
  - [ ] candidate evaluate
  - [ ] quality gate の結果確認
  - [ ] 実行結果をログまたは report として確認できる
- [ ] CI から production promotion は行わない

### 2. Reproducibility checks

- [ ] random seed / test split が固定・追跡できることを確認する
- [ ] training data hash が metadata と一致することを自動確認する
- [ ] code commit SHA / hyperparameters / dependency version / metrics を追跡できることを確認する
- [ ] 同じ fixture・設定から再実行したとき、少なくとも学習条件と評価経路を説明・追跡できるテストを追加する
- [ ] 「完全に同じ数値」だけを再現性の定義にせず、由来を追えることも検証対象にする

### 3. Batch classification / drift report

- [ ] production model で JSONL を一括分類できるようにする
- [ ] `python -m gaiden_sieve drift --profile ai-gaiden --input incoming.jsonl` 相当を実装する
- [ ] v0 の簡単な drift 指標を出す
  - [ ] 平均予測確率
  - [ ] uncertain の割合
  - [ ] source ごとの予測分布
  - [ ] 可能なら未知語 / 語彙外傾向を小さく観測する
- [ ] report を `reports/<profile>/` 等へ保存できる形にする
- [ ] drift は「異常を断定する仕組み」ではなく、再学習候補に気づくための観測値として扱う

### 4. Uncertain queue

- [ ] batch classification のうち `uncertain` を `uncertain.jsonl` へ切り出せるようにする
- [ ] 元記事を後から教師データ候補へ戻せるよう、少なくとも id / source / title / summary / probability / classified_at / model version を残す
- [ ] LLM 自動レビューはまだ接続しない
- [ ] active learning の入口として README に流れを説明する

### 5. Tests / docs

- [ ] GitHub Actions 相当のコマンドをローカルでも再現できる
- [ ] drift report の集計値を fixture でテストする
- [ ] uncertain queue が high / low confidence item を誤って含めないことをテストする
- [ ] profile ごとのしきい値が batch / drift / queue に反映されることをテストする
- [ ] README に Phase 3 の MLOps loop と学習ポイントを追記する

## 学習時に確認すること

1. **CI は何を守るのか**
   - production を自動更新するためではなく、コード・学習・評価経路が壊れていないことを継続確認する。

2. **再現性とは何か**
   - seed 固定だけでなく、data / code / config / metrics / artifact の対応関係を後から辿れること。

3. **drift は何を意味するのか**
   - コードが正常でも、入力記事の語彙や分布が変わればモデル性能が落ちうる。v0 では簡単な観測値から変化に気づく。

4. **uncertain queue はなぜ必要か**
   - 全件を高価な外部モデルへ送らず、分類器が苦手な例だけを将来の LLM review / human review / retraining へ戻す入口にする。

## 制約

- 実RSSにはまだ接続しない
- AI外電の既存フローにはまだ接続しない
- 自動掲載には接続しない
- CI から production model を自動 promote しない
- LLM 自動レビューは実装しない
- scheduled retraining はまだ実装しない
- 高度な drift detector / monitoring service は導入しない
- Phase 4（実RSS / shadow mode）へ広げない
- `main` へ直接実装せず、専用ブランチ + PR で進める

## 完了条件

- [ ] `pytest` が通る
- [ ] GitHub Actions で test → train → evaluate が通る
- [ ] CI が production を変更しない
- [ ] production model で fixture JSONL を一括分類できる
- [ ] 簡単な drift report を生成できる
- [ ] uncertain item を queue へ保存できる
- [ ] model / data / code / config / metrics の対応関係を追跡できる
- [ ] Phase 4 の実RSS shadow modeへ進める状態になっている

## 正本

- `AGENTS.md`
- `DESIGN.md` §10, §11, §13–§16, §18, §19
- Phase 2: #4

**Comment by yo4e** [comment]
Phase 3 は PR #7 で実装・検証し、main へマージしました。

- GitHub Actions: pytest → train → evaluate → reproducibility verify → report
- JSONL batch classification
- drift report
- uncertain.jsonl queue
- README / pytest 更新
- CI から production promotion しないことを確認

PR #7: https://github.com/yo4e/gaiden-sieve/pull/7
Merge commit: e24072954c241adbfffd943e96d367b6cd947317

次の段階は DESIGN.md の Phase 4: 実RSS item を使った shadow mode 接続実験です。
